InMemoryFileSys/Directory.cs

```diff
@@ -10,139 +12,124 @@ public class Directory : IDirectory
     public DateTime CreationDate { get; set; }
 
     // Public property to access the singleton instance
-    public static Directory Root => _root.Value;    
+    public static Directory Root => _root.Value;   
+    
     /// <inheritdoc/>
-    public int? Size
-    {
-        get
-        {
-            int totalSize = 0;
-            foreach (var entry in _entries)
-            {
-                if (entry is IFile file)
-                {
-                    totalSize += file.Size ?? 0;
-                }
-                else if (entry is IDirectory directory)
-                {
-                    totalSize += directory.Size ?? 0;
-                }
-            }
+    public int? Size { get; private set; }
+   
+    public string Path { get; }
 
-            return totalSize;
-        }
-    }
-    
-    private List<IFileSystemEntry> _entries { get; }
+    private readonly Dictionary<string, IFileSystemEntry> _entries;
     
     // Singleton instance
-    private static readonly Lazy<Directory> _root = new Lazy<Directory>(() => new Directory("/", new SystemClock()));
-    
+    // Singleton instance
+    private static readonly Lazy<Directory> _root = 
+        new(() => new Directory(rootName, new SystemClock(), null));
+
     private readonly IClock _clock;
 
-    private Directory(string? name, IClock clock)
+    private readonly Directory? _parentDirectory;
+    
+    private const string slash = "/";
+
+    private const string rootName = "/";
```

Comment:
equal to slash

Reply:
ok, I know, but it felt strange to use slash as name so I created two, but I removed it